clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7825,7 +7829,7 @@ void Clang::AddClangCLArgs(const ArgList &Args, types::ID InputType,
 
   const Driver &D = getToolChain().getDriver();
   EHFlags EH = parseClangCLEHFlags(D, Args);
-  if (!isNVPTX && (EH.Synch || EH.Asynch)) {
+  if (!isNVPTX && !isSPIR && (EH.Synch || EH.Asynch)) {
```

Comment:
Might also need to account for CUDA/HIP

Reply:
This is testing the triple, not the offload kind so I think we are OK here.